api/envoy/api/v2/auth/cert.proto

```diff
@@ -289,6 +289,16 @@ message CertificateValidationContext {
   // An optional list of Subject Alternative name matchers. Envoy will verify that the
   // Subject Alternative Name of the presented certificate matches one of the specified matches.
   //
+  // When a certificate has wildcard DNS SAN entries, to match a specific client, it should be
+  // configured with exact match type in the string matcher. For example if the certificate has
+  // "\*.example.com" as DNS service entry, to allow only "api.example.com", it should be configured
+  // as shown below.
+  //
+  // .. code-block:: yaml
+  //
+  // match_subject_alt_names:
+  //   exact: "app.example.com"
+  //
```

Comment:
These documentation changes are great at explaining the necessary configuration for matching with wildcard SAN entries!

A few minor nits:
- link/add ref to StringMatcher/exact_match (e.g. :ref:\`string matcher <envoy_api_msg_type.matcher.StringMatcher>\`)
- I'm guessing "DNS service entry" should be "DNS SAN entry"?
- "app.example.com" should be "api.example.com"